chain-impl-mockchain/src/evm/mod.rs

```diff
@@ -47,133 +48,66 @@ pub enum EvmTransaction {
         value: U256,
         data: ByteCode,
         gas_limit: u64,
-        access_list: AccessList,
+        access_list: Vec<(Address, Vec<Key>)>,
     },
 }
 
-#[cfg(feature = "evm")]
-impl From<&EvmTransaction> for u8 {
-    fn from(other: &EvmTransaction) -> Self {
-        use EvmTransaction::*;
-        match other {
-            Create { .. } => 0,
-            Create2 { .. } => 1,
-            Call { .. } => 2,
-        }
-    }
-}
-
-#[cfg(feature = "evm")]
-impl Decodable for EvmTransaction {
-    fn decode(rlp: &Rlp<'_>) -> Result<Self, DecoderError> {
-        fn decode_tx_create(rlp: &Rlp<'_>) -> Result<EvmTransaction, DecoderError> {
-            Ok(EvmTransaction::Create {
-                caller: rlp.val_at(1)?,
-                value: rlp.val_at(2)?,
-                init_code: rlp.val_at(3)?,
-                gas_limit: rlp.val_at(4)?,
-                access_list: rlp.list_at(5)?,
-            })
-        }
-        fn decode_tx_create2(rlp: &Rlp<'_>) -> Result<EvmTransaction, DecoderError> {
-            Ok(EvmTransaction::Create2 {
-                caller: rlp.val_at(1)?,
-                value: rlp.val_at(2)?,
-                init_code: rlp.val_at(3)?,
-                salt: rlp.val_at(4)?,
-                gas_limit: rlp.val_at(5)?,
-                access_list: rlp.list_at(6)?,
-            })
-        }
-        fn decode_tx_call(rlp: &Rlp<'_>) -> Result<EvmTransaction, DecoderError> {
-            Ok(EvmTransaction::Call {
-                caller: rlp.val_at(1)?,
-                address: rlp.val_at(2)?,
-                value: rlp.val_at(3)?,
-                data: rlp.val_at(4)?,
-                gas_limit: rlp.val_at(5)?,
-                access_list: rlp.list_at(6)?,
-            })
-        }
-
-        match rlp.val_at(0)? {
-            0u8 => decode_tx_create(rlp),
-            1u8 => decode_tx_create2(rlp),
-            2u8 => decode_tx_call(rlp),
-            _ => Err(DecoderError::Custom("invalid evm transaction")),
-        }
-    }
-}
-
-#[cfg(feature = "evm")]
-impl Encodable for EvmTransaction {
-    fn rlp_append(&self, s: &mut RlpStream) {
-        use EvmTransaction::*;
+impl EvmTransaction {
+    /// Serialize the contract into a `ByteBuilder`.
+    pub fn serialize_in(&self, _bb: ByteBuilder<Self>) -> ByteBuilder<Self> {
```

Comment:
I would prefer to implement `Serialize` trait, it will allow to use `serialization_bijection` function in tests and it will follow to the full serde interface. (as `Deserialize` trait has been implemented)

Reply:
I restored the `Serialize` trait, and kept the RLP encoding because it handles things like `U256` values and list of items better than we were doing it. If anything, I would move the EvmTransaction type into `chain-evm` to clean up dependencies a bit, but that would be in a future PR.